minio/minio
# lifecycle: Fix rules filtering

## Description
A rule with a tag prefix can be applied to an object which does not have
the tag associated to it. This commit fixes it by simplifying the filter
code.

## Motivation and Context
Fix applying rules with tag filter on objects without tag

## How to test this PR?
1) Run two MinIO server, one primary the second will be for tiering
2) Create 'testbucket' in both clusters
3) ./mc admin tier add minio myminio TIERS3 --endpoint http://localhost:9001 --access-key minioadmin --secret-key minioadmin --bucket testbucket
4) ./mc ilm add --transition-days 0 --tags tier=TIERS3 --storage-class TIERS3  myminio/testbucket
5) Upload an object and check if it is tiered or not after few minutes using 'mc stat' command

## Types of changes

## Checklist:
